gradle/root/coverage.gradle

```diff
@@ -1,15 +1,27 @@
-apply plugin: 'jacoco'
+//TODO: Standard JaCoCo coverage doesn't work in Android module
+task mockitoCoverage(type: JacocoReport) {
 
-// TODO later include subprojects, possible ideas https://gist.github.com/aalmiray/e6f54aa4b3803be0bcac
+    allprojects { currentProject ->
+        plugins.withId("java") {
+            mockitoCoverage.sourceSets currentProject.sourceSets.main
 
-jacoco {
-    toolVersion = "0.7.9"
-}
+            apply plugin: "jacoco"
+
+            jacoco {
+                toolVersion = '0.7.9' //Gradle 4.2.1 uses 0.7.8 by default
 
-task mockitoCoverage(type: JacocoReport, dependsOn: "test") {
-    executionData files("${buildDir}/jacoco/test.exec")
+                if (currentProject != rootProject) {   //already automatically enhanced in mockito main project as "java" plugin was applied before
```

Comment:
@mockitoguy Do you have a good idea how to get rid of it without changing the main Gradle configuration in Mockito to much?

Subtasks doesn't have `Test` tasks enhances automatically due to the applying order.

Reply:
Not sure. I don't fully see the problem with ordering.

Reply:
According [documentation](https://docs.gradle.org/current/userguide/jacoco_plugin.html#sec:jacoco_specific_task_configuration):
> The JaCoCo plugin adds a JacocoTaskExtension extension to all tasks of type Test. This extension allows the configuration of the JaCoCo specific properties of the test task.

It doesn't happen for subprojects here. I needed to call "applyTo" manually. I didn't see it in multiprojects with "normal" empty parent/root project.